### PR TITLE
Retires for task scheduler

### PR DESCRIPTION
**Pull request by runk**

### CODE REVIEW

File: lib/taskRunner.js [hunk 1]
```diff
@@ -34,97 +34,165 @@ util.inherits(TaskRunner, EventEmitter);
...
-        case 'testsDone':
-          runResults.failedCount = m.results.failedCount;
-          runResults.specResults = m.results.specResults;
-          break;
-      }
-    })
-    .on('error', function(err) {
-      taskLogger.flush();
-      deferred.reject(err);
+  var deferred = q.defer();
```
Comment (antklim) [inline lib/taskRunner.js:37]: Why the half of the lines end with `;` and the other half with nothing?

Comment (runk) [inline lib/taskRunner.js:37]: Because it's javascript? Will fix it.

File: lib/taskRunner.js [hunk 2]
```diff
@@ -34,97 +34,165 @@ util.inherits(TaskRunner, EventEmitter);
...
+    taskLogger.log(data);
+  });
+
+  // stderr pipe
+  childProcess.stderr.on('data', function(data) {
+    taskLogger.log(data);
+  });
+
+  childProcess.on('message', function(m) {
+    // console.log('zzz msg', self.task.taskId, m.event);
```
Comment (antklim) [inline lib/taskRunner.js:127]: Remove?

Comment (runk) [inline lib/taskRunner.js:127]: I want to keep my debugging console.logs for time being. I'll send a change request to original protractor repo once my changes tested

**Pull request merged**